CONTRIBUTING.md

```diff
@@ -1,143 +1,269 @@
+    * [Manual method: Stashing](#manual-method-stashing)
+  * [Method 2 - Fixup commits](#method-2---fixup-commits)
+    * [Manual method](#manual-method)
+    * [Automatic method](#automatic-method)
+* [PR Policy](#pr-policy)
+* [Formatting](#formatting)
+* [Patch Notes](#patch-notes)
+* [Obfuscation Helpers](#obfuscation-helpers)
+* [Configuration files](#configuration-files)
+  * [PaperConfig example](#paperconfig-example)
+  * [PaperWorldConfig example](#paperworldconfig-example)
+* [Frequently Asked Questions](#frequently-asked-questions)
+  * [I can't find the NMS file I need!](#i-cant-find-the-nms-file-i-need)
+  * [Where can I learn how to name method/field?](#where-can-i-learn-how-to-name-methodfield)
+  * [My commit doesn't need a build, what do I do?](#my-commit-doesnt-need-a-build-what-do-i-do)
+  * [Patching and building is *really* slow, what can I do?](#patching-and-building-is-really-slow-what-can-i-do)
+  * [I wrote some API, how do I use it in Paper-Server?](#i-wrote-some-api-how-do-i-use-it-in-paper-server)
+
+<!-- vim-markdown-toc -->
 
 ## Use a Personal Fork and not Organization
-Paper will routinely modify your PR for you to rebase it on latest or take care of any minor nitpicks that we can
-just solve for you instead of making you go back for a trivial item.
 
-However, if you use an Organization for the PR, this prevents Paper from doing this, and will require us to manually
-merge it and close the PR instead of it being marked as the purple merged.
+Paper will routinely modify your PR, whether it's a quick rebase or to take care
+of any minor nitpicks we might have. Often, it's better for us to solve these
+problems for you than make you go back and forth trying to fix it yourself.
 
-We much prefer to have merged PR's show as merged, so please do not use organizations repo's for PR as GitHub sadly
-does not support maintainers pushing to them.
+Unfortunately, if you use an organization for your PR, it prevents Paper from
+modifying it. This requires us to manually merge your PR, resulting in us
+closing the PR instead of marking it as merged.
 
-See [https://github.com/isaacs/github/issues/1681](https://github.com/isaacs/github/issues/1681)
+We much prefer to have PRs show as merged, so please do not use repositories
+on organizations for PRs.
 
-## Understanding Patches
-Patches to Paper are very simple, but center around the directories 'Paper-API' and 'Paper-Server'
+See <https://github.com/isaacs/github/issues/1681> for more information on the
+issue.
 
-Assuming you already have forked the repository:
+## Requirements
 
-1. Pull the latest changes from the main repository
-1. Type `./paper patch` in git bash to apply the changes from upstream
-1. cd into `Paper-Server` for server changes, and `Paper-API` for API changes
+To get started with PRing changes, you'll need the following software, most of
+which are all available in (most) package managers such as `apt` (Debian /
```

Comment:
```suggestion
which is available in (most) package managers such as `apt` (Debian /
```

Reply:
`most of which is` when talking about a mass noun (according to [Oxford dictionaries](https://www.lexico.com/definition/software) anyway) is not correct either, I believe. I'll wait for someone else to kick in with a correction if I'm wrong.

Reply:
I think you're wrong. "software, most of which are available" sounds off-putting. Software is, not "software are".

Reply:
I just reworded it as a whole to avoid potential pluralisation.

Reply:
"Mass-noun" is just a fancy term for uncountable.. I also don't see anything on that page that suggests you'd treat it as a normal plural (with "are") 👀